### ISSUE TITLE
QUESTION: access parameter from parent root

### ISSUE THREAD

**Issue by gbaccetta** [Milad-Akarie/auto_route_library#473]
Hi and thantk you for your hard work.
It's amazing how much time and boilerplate I'm saving thanks to your package.

I've a question that is maybe a more general flutter navigation question.
I've a Parent Route with a nested autoTabRouter.
So imagine something like this : `/home/product/:id/` for the parent route 
and : `/home/product/:id/tab1/reviews`, `/home/product/:id/tab2/alternatives` ... and so on

Is it possible to access the `id `parameter of the parent router from one of the tab routers or should I pass it down to each of the tab routers screen that need it ?

On solution I thought of is to parse the path of the current tab route, but I'm not sure this is the fastest way to achieve that

**Comment by Milad-Akarie** [comment]
Hello, I'm glad you like the lib.
I'm replying from mobile so I'm keeping it short.

Each route has a scoped RouteData object
You can access by calling RouteData.of()
Or context.routeData

RouteData objects are paren aware 
So calling context.routeData?.parant will get you the parent route

Nested Routing Controller also has an associated RouteData object so it'd be easier for you to just go
 context.tabsRouter.routeData
 from inside of your tabs

Let me know if you have more questions

**Comment by gbaccetta** [comment]
Ok, that's what I thought.
So there are basically trhee ways to go:
1. `AutoRouter.of(context).parent()!.routeData!.pathParams.getString('id');`
2. `context.routeData.parent!.parent!.pathParams.getString('id');`
3. (**BEST**) `context.tabsRouter.routeData!.pathParams.getString('id');`

notice that in the second line I had to call twice parent from the context.routeData beacause I got the TabRouter from the first call and the stackrouter from the second one.

Of course your suggestion makes for the neatest and shortest way to access it (the third)!
Thank you !
